From 81ec548f7e20b6ad57e656e6a389b71f7479d933 Mon Sep 17 00:00:00 2001
From: Aurelian Zanoschi <aurelian17@gmail.com>
Date: Fri, 22 Jul 2016 23:13:44 +0300
Subject: [PATCH] RF24Network Fix cross compilation

In the current shape, the Makefile provided with the RF24Network
package cannot be cross compiled. PREFIX was provided as an
environment variable.

Signed-off-by: Aurelian Zanoschi <aurelian17@gmail.com>
---
 Makefile | 29 ++++++++++++++---------------
 1 file changed, 14 insertions(+), 15 deletions(-)

diff --git a/Makefile b/Makefile
index 59a41b6..7a32473 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@
 # use make all and mak install to install the library 
 # You can change the install directory by editing the LIBDIR line
 #
-PREFIX=/usr/local
+#PREFIX=/usr/local
 
 # Library parameters
 # where to put the lib
@@ -23,22 +23,22 @@ LIBNAME_RFN=$(LIB_RFN).so.1.0
 
 HEADER_DIR=${PREFIX}/include/RF24Network
 
-ARCH=armv6zk
-ifeq "$(shell uname -m)" "armv7l"
-ARCH=armv7-a
-endif
+#ARCH=armv6zk
+#ifeq "$(shell uname -m)" "armv7l"
+#ARCH=armv7-a
+#endif
 
 # Detect the Raspberry Pi from cpuinfo
 #Count the matches for BCM2708 or BCM2709 in cpuinfo
-RPI=$(shell cat /proc/cpuinfo | grep Hardware | grep -c BCM2708)
-ifneq "${RPI}" "1"
-RPI=$(shell cat /proc/cpuinfo | grep Hardware | grep -c BCM2709)
-endif
+#RPI=$(shell cat /proc/cpuinfo | grep Hardware | grep -c BCM2708)
+#ifneq "${RPI}" "1"
+#RPI=$(shell cat /proc/cpuinfo | grep Hardware | grep -c BCM2709)
+#endif
 
-ifeq "$(RPI)" "1"
+#ifeq "$(RPI)" "1"
 # The recommended compiler flags for the Raspberry Pi
-CCFLAGS=-Ofast -mfpu=vfp -mfloat-abi=hard -march=$(ARCH) -mtune=arm1176jzf-s -std=c++0x
-endif
+CCFLAGS=-Ofast -std=c++0x $(CXXFLAGS)
+#endif
 
 # make all
 # reinstall the library after each recompilation
@@ -46,11 +46,11 @@ all: librf24network
 
 # Make the library
 librf24network: RF24Network.o
-	g++ -shared -Wl,-soname,$@.so.1 ${CCFLAGS} -o ${LIBNAME_RFN} $^ -lrf24-bcm
+	$(CXX) -shared -Wl,-soname,$@.so.1 ${CCFLAGS} -o ${LIBNAME_RFN} $^ -lrf24-bcm
 
 # Library parts
 RF24Network.o: RF24Network.cpp
-	g++ -Wall -fPIC ${CCFLAGS} -c $^
+	$(CXX) -Wall -fPIC ${CCFLAGS} -c $^
 
 # clear build files
 clean:
@@ -66,7 +66,6 @@ install-libs:
 	@install -m 0755 ${LIBNAME_RFN} ${LIBDIR}
 	@ln -sf ${LIBDIR}/${LIBNAME_RFN} ${LIBDIR}/${LIB_RFN}.so.1
 	@ln -sf ${LIBDIR}/${LIBNAME_RFN} ${LIBDIR}/${LIB_RFN}.so
-	@ldconfig
 
 install-headers:
 	@echo "[Installing Headers]"
-- 
1.9.1

